tests/SuccessfulCheckout.spec.ts

```diff
@@ -0,0 +1,55 @@
+import {expect, test} from '../fixtures/fixtures';
+import {LoginPage} from '../pages/login.page';
+import * as dotenv from 'dotenv';
+import { config } from '../env.config';
+dotenv.config();
+
+test('Verify successful checkout with logged in user', async ({ loggedInPage, productPage, cartPage, checkoutPage }) => {
+
+  await loggedInPage.openProduct(' Combination Pliers ');
+
+  await expect(productPage.page).toHaveURL(/product/);
+
+  const actualProductName = await productPage.productName.innerText();
+  const actualProductPriceText = await productPage.productPrice.innerText();
+  const actualProductPrice = actualProductPriceText ? actualProductPriceText.replace('$', '') : '';
+
+  await productPage.addToCart();
+
+  await loggedInPage.header.proceedToCheckout();
+  await cartPage.verifyProductDetails(actualProductName, actualProductPrice);
+  await cartPage.verifyTotalPrice(actualProductPrice);
+
+  await cartPage.proceedToCheckout();
+
+  const loginPageOnCheckout = new LoginPage(checkoutPage.page); 
+
+  await loginPageOnCheckout.login(config.USER_EMAIL!, config.USER_PASSWORD!);
+
+  await checkoutPage.expectUserIsLoggedIn();
+
+  const currentDate = new Date();
```

Comment:
Let's create a separate method in the `utils` folder to hide this logic from tests